src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegment.cs

```diff
@@ -59,7 +59,7 @@ public void SetOwnedMemory(byte[] arrayPoolBuffer)
             AvailableMemory = arrayPoolBuffer;
         }
 
-        public void ResetMemory()
+        public void ResetMemory(bool preserveIndex = false)
```

Comment:
Nit:

Maybe add a `ClearMemory()` method and rename this to `Reset()` and have it call `ClearMemory()`. Because this is doing more than just resetting memory, it's resetting the `RunningIndex` and the `_next` pointer too. It might make sense for `ClearMemory()` to keep the `ResetMemory()` name.

Reply:
👍 I was thinking along those lines when writing it. Good to know we're thinking that same thing.